winml/lib/Api/LearningModelSession.h

```diff
@@ -114,11 +120,13 @@ struct LearningModelSession : LearningModelSessionT<LearningModelSession> {
 
   // Synchronization
   CWinMLLock session_creation_lock_;
-  CWinMLLock evaluate_lock_;
+  static CWinMLLock dml_ep_lock_;
```

Comment:
This is a layering violation, as winml now needs to know about the dml ep's threading behavior.

This should be moved into the dml ep... is there a bug to track that work?

Reply:
Bug 25583152: DML EP is not thread safety